File: day1/part1/solution.py
import os

def solve(rotations):
    """
    Calculates the password based on a list of rotation instructions.
    """
    current_position = 50
    zero_count = 0

    for line in rotations:
        line = line.strip()
        if not line:
            continue

        direction = line[0]
        distance = int(line[1:])

        if direction == 'R':
            current_position += distance
        elif direction == 'L':
            current_position -= distance
        
        current_position %= 100

        if current_position == 0:
            zero_count += 1
    
    return zero_count

def main():
    """
    Reads the input file and prints the solution.
    """
    input_file_path = os.path.join(os.path.dirname(__file__), 'input1.txt')
    
    with open(input_file_path, 'r') as f:
        rotations = f.readlines()
        
    password = solve(rotations)
    print(f"The password is: {password}")

if __name__ == "__main__":
    main()

File: day1/part1/solution.js
const fs = require('fs');
const path = require('path');

function solve(input) {
    let currentPosition = 50;
    let zeroCount = 0;

    const rotations = input.split('\n').filter(line => line.trim() !== '');

    for (const rotation of rotations) {
        const direction = rotation[0];
        const distance = parseInt(rotation.substring(1), 10);

        if (direction === 'R') {
            currentPosition += distance;
        } else if (direction === 'L') {
            currentPosition -= distance;
        }

        // Handle the circular dial, ensuring the result is always positive
        currentPosition = ((currentPosition % 100) + 100) % 100;

        if (currentPosition === 0) {
            zeroCount++;
        }
    }

    return zeroCount;
}

function main() {
    const inputFilePath = path.join(__dirname, 'input1.txt');
    const input = fs.readFileSync(inputFilePath, 'utf8');
    const password = solve(input);
    console.log(`The password is: ${password}`);
}

if (require.main === module) {
    main();
}
